packages/client/lib/sync/sync.ts

```diff
@@ -47,6 +52,7 @@ export abstract class Synchronizer extends EventEmitter {
 
     this.pool = options.pool
     this.chain = options.chain
+    this.stateDB = options.stateDB
```

Comment:
I don't think we need `stateDB` here, since it does not seem we use it anywhere?

Reply:
I put it there and not directly in `FullSynchronizer` since I thought this might be more flexible if we want to use `stateDB` in other synchronizers. Do you want me to move over to `FullSynchronizer` directly? Don't have a strong opinion on this myself, think these are all things which can be easily evolved and adopted further down the road.

Reply:
I have no strong opinion either, it is OK to keep it here for now, if we focus on the light client then we can remove it if it turns out it is unnecessary.